### PR TITLE
Introduce linear backoff for client sleep time

### LINKED ISSUE #97: [Investigation] Log streaming
Throttling limits from blob size; linear backoff for retries

### PR DESCRIPTION
**Pull request by ehotinger**
Tried to DOS one of our Azure Storage accounts, couldn't make it happen with multiple clients on a 0 second delay. This reduces the time before we query for logs again to 1 second, which will be a partial improvement to perceived latency. I don't see any reason to introduce more complex transient fault handling for now - AZ Storage handles it pretty seamlessly as far as I can tell.

Fixes #97

### CODE REVIEW

**Comment by NatalieGong** [comment]
I actually have no idea how will the sleep time effect the whole system. Can you explain?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def _stream_logs(byte_size,
         # If no new data available but not complete, sleep before trying
         # to process additional data.
         if (_blob_is_not_complete(metadata) and start >= available):
-            time.sleep(5)
+            time.sleep(1)
```
Comment (northtyphoon) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py]: storage rps limit per account is 20k, it's probably ok considering the payload size. need to watch if impacting the requests on other artifacts on the same account. The worst case is when no update in 30 minutes if the build fails. We may consider adding a check on build status every minute or something like that.

**Review by northtyphoon — APPROVED** [review]

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py (hunk 1 above)
Comment (ehotinger) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py]: @northtyphoon I thought about this again and decided to introduce a linear backoff. I tried this several times and we just end up hitting storage too frequently and the client doesn't make any progress. I think something like this is a better intermediate for the worst case scenarios. The curse of pull models. :( Please review again.

It'll do this:

- 3 tries @ 1 sec
- 3 tries @ 2 
- 3 tries @ 4
- 3 tries @ 8
- infinite tries @ 15 (MAX) until timeout

/cc @sajayantony

**Review by djyou — APPROVED** [review]
Naming convention `a_b_c`.

**Review by northtyphoon — APPROVED** [review]
👍

**Pull request merged**